src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -177,8 +183,8 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
                                     throw new InvalidOperationException($"Can't find the latest version of {purl}");;
 
             // Construct a new PackageURL that's guaranteed to have a version, the latest version is used if no version was provided.
-            PackageURL purlWithVersion = !string.IsNullOrWhiteSpace(purl.Version) ? 
-                purl : new PackageURL(purl.Type, purl.Namespace, purl.Name, latestVersion, purl.Qualifiers, purl.Subpath);
+            string guaranteedVersion = !string.IsNullOrWhiteSpace(purl.Version) ? purl.Version.ToLowerInvariant() : latestVersion;
```

Comment:
I don't think this is necessary. The spec indicates that version checking is case insensitive. Not that it requires lowercase. 

Can you provide an example of when this change is required?